### PR TITLE
drivers: usart for LPC devices interrupt enabled

### LINKED ISSUE #6039: Implement interrupt driven USART on LPC54114
Currently the usart shim driver for LPC54114 only supports polling mode; add support for interrupt-driven mode. This is needed to support the console shell.

### PR DESCRIPTION
**Pull request by agansari**
LPC54xxx and LPC55xxx USART interrupts enables.
This will allow to use console on LPC devices.

Fixes #6039

Tested with samples/subsys/console/echo

### CODE REVIEW

File: drivers/serial/usart_mcux_lpc.c [hunk 1]
```diff
@@ -83,6 +88,158 @@ static int usart_mcux_lpc_err_check(struct device *dev)
...
+
+static int usart_mcux_lpc_fifo_read(struct device *dev, u8_t *rx_data,
+			       const int len)
+{
+	const struct usart_mcux_lpc_config *config = dev->config->config_info;
+	u8_t num_rx = 0U;
+
+	while ((len - num_rx > 0) &&
+	       (USART_GetStatusFlags(config->base)
+		& kUSART_RxFifoFullFlag)) {
```
Comment (MaureenHelm) [inline drivers/serial/usart_mcux_lpc.c]: Wouldn't `kUSART_RxFifoNotEmptyFlag` be better here?

File: drivers/serial/usart_mcux_lpc.c [hunk 2]
```diff
@@ -83,6 +88,158 @@ static int usart_mcux_lpc_err_check(struct device *dev)
...
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+static int usart_mcux_lpc_fifo_fill(struct device *dev, const u8_t *tx_data,
+			       int len)
+{
+	const struct usart_mcux_lpc_config *config = dev->config->config_info;
+	u8_t num_tx = 0U;
+
+	while ((len - num_tx > 0) &&
+	       (USART_GetStatusFlags(config->base)
+		& kUSART_TxFifoEmptyFlag)) {
```
Comment (MaureenHelm) [inline drivers/serial/usart_mcux_lpc.c]: Wouldn't `kUSART_TxFifoNotFullFlag` be better here?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/serial/usart_mcux_lpc.c (hunk 2 above)
Comment (agansari) [reply drivers/serial/usart_mcux_lpc.c]: @MaureenHelm you are right

Comment (MaureenHelm) [reply drivers/serial/usart_mcux_lpc.c]: You updated fifo_read to use `kUSART_RxFifoNotEmptyFlag`, but you didn't update fifo_fill to use `kUSART_TxFifoNotFullFlag` yet.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
I tested `samples/subsys/console/echo` on lpcxpresso54114_m4 and it works.

File: drivers/serial/usart_mcux_lpc.c [hunk 3]
```diff
@@ -51,8 +56,8 @@ static void usart_mcux_lpc_poll_out(struct device *dev,
 	const struct usart_mcux_lpc_config *config = dev->config->config_info;
 
 	/* Wait until space is available in TX FIFO */
-	while (!(USART_GetStatusFlags(config->base) & kUSART_TxFifoEmptyFlag)) {
-	}
+	while (!(USART_GetStatusFlags(config->base) & kUSART_TxFifoEmptyFlag))
+		;
```
Comment (MaureenHelm) [inline drivers/serial/usart_mcux_lpc.c]: extraneous change after rebasing? #16616 recently added braces around all if/while statements

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**